Source/WebKit/UIProcess/WebCookieManagerProxy.cpp

```diff
@@ -49,7 +49,8 @@ WebCookieManagerProxy::~WebCookieManagerProxy()
 {
     if (m_networkProcess)
         m_networkProcess->removeMessageReceiver(Messages::WebCookieManagerProxy::messageReceiverName());
-    ASSERT(m_cookieObservers.isEmpty());
+    if (!m_cookieObservers.isEmpty())
+        RELEASE_LOG(Storage, "WebCookieManagerProxy::~WebCookieManagerProxy %u cookie observers will be invalidated", m_cookieObservers.size());
```

Comment:
Doesn't this mean that the API client may still have a registered cookie observer and yet no longer receive any notifications because the WebCookieManagerProxy was destroyed? Wouldn't this be a bug?

Seems like if the clients is monitoring cookies, they should be getting updates until they unregister their observer.

Reply:
Yes, just like the failed test. If client has a registered cookie observer but WebsiteDataStore is destroyed, the observer will not get any notification. It looks reasonable to me that client does not get notification about cookies of an ephemeral session if the session is destroyed?
I am not sure how this API is supposed to work originally, at least according to current implementation, the assertion seems to not hold.